### ISSUE TITLE
Creates types incompatible with typescript 3.x

### ISSUE THREAD

**Issue by Narretz** [francescov1/mongoose-tsgen#60]
See this repo: https://github.com/Narretz/mongoose-tsgen-issues/tree/typescript (make sure to check out the **typescript** branch)

I don't think it's documented explicitly which typescript versions are supported, but the package.json itself requires ^3.9.7. However, if your own code is requiring typescript 3.9.7, it won't work, because te generated type will throw a bunch of errors:

```
src/interfaces/mongoose.gen.ts:268:35 - error TS2344: Type 'Unarray<ResultType>' does not satisfy the constraint 'Document<any, any>'.
  Type '{}' is missing the following properties from type 'Document<any, any>': $getAllSubdocs, $ignore, $isDefault, $isDeleted, and 48 more.

268         ? Array<PopulatedDocument<Unarray<ResultType>, T>>
                                      ~~~~~~~~~~~~~~~~~~~

src/interfaces/mongoose.gen.ts:270:29 - error TS2344: Type 'Unarray<ResultType>' does not satisfy the constraint 'Document<any, any>'.
  Type 'unknown' is not assignable to type 'Document<any, any>'.

270         ? PopulatedDocument<Unarray<ResultType>, T>
                                ~~~~~~~~~~~~~~~~~~~

src/interfaces/mongoose.gen.ts:281:35 - error TS2344: Type 'Unarray<ResultType>' does not satisfy the constraint 'Document<any, any>'.

281         ? Array<PopulatedDocument<Unarray<ResultType>, T>>
                                      ~~~~~~~~~~~~~~~~~~~

src/interfaces/mongoose.gen.ts:283:29 - error TS2344: Type 'Unarray<ResultType>' does not satisfy the constraint 'Document<any, any>'.

283         ? PopulatedDocument<Unarray<ResultType>, T>
                                ~~~~~~~~~~~~~~~~~~~
```

**Comment by francescov1** [comment]
Good catch, I'll look into this fix over the weekend! For the time being, you should be able to pass the `--no-populate-overload` flag to avoid augmenting populations (if those errors above are the only issue). Will get back to you soon!

**Comment by francescov1** [comment]
Fixed in 8.3.5! Added Typescript as a peer dependency.
